File: api/filing/models/tcs_model.py
"""TCS (Tax Collected at Source) model."""
from datetime import date
from .api_base_model import ApiBaseModel


class TCSModel(ApiBaseModel):
    tcs_id: int | None = None
    filing_id: int | None = None
    collector_name: str | None = None
    tan: str | None = None
    nature_of_collection: str | None = None
    amount_collected: float | str | int | None = None
    tax_collected: float | str | int | None = None
    tcs_certificate_number: str | None = None
    quarter: str | None = None
    year_of_collection: date | None = None
    tax_credit_claimed: float | str | int | None = None

File: web/filing/models/tax-credits/tcs-model.ts
/**
 * TCS Model
 * Tax Collected at Source
 */
export interface TCSModel {
  tcsId?: number | null
  filingId: number | null
  collectorName: string
  tan: string
  natureOfCollection?: string | null
  amountCollected?: number | null
  taxCollected?: number | null
  tcsCertificateNumber?: string | null
  quarter?: string | null
  yearOfCollection?: Date | null
  taxCreditClaimed?: number | null
}

export interface TCSResponse {
  data: TCSModel
  message?: string
}

export interface TCSListResponse {
  data: TCSModel[]
  message?: string
}

export interface TCSEntryWidgetProps {
    isOpen: boolean
    onClose: () => void
    filingId: number | null
    entityId?: number | string
    initialData?: Record<string, any> | null
    onSuccess?: (action: 'created' | 'updated', data: any) => void
    errorField?: string | null
    errorMessage?: string | null  // Error message to display for the field
    fieldErrors?: Record<string, string> | null  // Multiple field errors: { fieldName: errorMessage }
}
